In [1]:
import pandas as pd

In [2]:
train_df = pd.read_excel('01 Train Data.xlsx')
test_df = pd.read_excel('02 Test Data.xlsx')

In [3]:
train_df.head()

First Name           Email ID  Quantity  Price Tier             Ticket Type  \
0     ANIKET     aniket@xyz.com         1         NaN  Art of Resume Building   
1  Dhanshree  dhanshree@xyz.com         1         NaN  Art of Resume Building   
2     Dhiraj     dhiraj@xyz.com         1         NaN  Art of Resume Building   
3      Pooja      pooja@xyz.com         1         NaN  Art of Resume Building   
4     Aayush     aayush@xyz.com         1         NaN  Art of Resume Building   

     Attendee #  Group  Order Type Currency  Total Paid  ...  Attendee Status  \
0  2.213855e+09    NaN  Free Order      USD           0  ...        Attending   
1  2.213859e+09    NaN  Free Order      USD           0  ...        Attending   
2  2.213862e+09    NaN  Free Order      USD           0  ...        Attending   
3  2.213988e+09    NaN  Free Order      USD           0  ...        Attending   
4  2.214567e+09    NaN  Free Order      USD           0  ...        Attending   

                                        College Name  \
0  D Y PATIL INSTITUTE OF MCA AND MANAGEMENT AKUR...   
1                   AP SHAH INSTITUTE OF TECHNOLOGY    
2       Don Bosco College of Engineering Fatorda Goa   
3           Pillai College of Engineering New Panvel   
4                               St Xavier's College    

        How did you come to know about this event?  \
0                                            Email   
1                                           Others   
2                                            Email   
3                                            Email   
4  Instagram | LinkedIn | Cloud Counselage Website   

  Specify in "Others" (how did you come to know about this event) Designation  \
0                                                NaN                 Students   
1                                            College                 Students   
2                                                NaN                 Students   
3                                                NaN                 Students   
4                                                NaN                 Students   

  Year of Graduation CGPA Speaking Skills ML Knowledge  Placement Status  
0                NaN  6.7               2            5            Placed  
1                NaN  8.2               3            2        Not placed  
2                NaN  6.5               4            3        Not placed  
3                NaN  8.7               2            5        Not placed  
4                NaN  9.1               3            5            Placed  

[5 rows x 23 columns]

In [4]:
train_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4894 entries, 0 to 4893
Data columns (total 23 columns):
 #   Column                                                           Non-Null Count  Dtype  
---  ------                                                           --------------  -----  
 0   First Name                                                       4894 non-null   object 
 1   Email ID                                                         4894 non-null   object 
 2   Quantity                                                         4894 non-null   int64  
 3   Price Tier                                                       0 non-null      float64
 4   Ticket Type                                                      4894 non-null   object 
 5   Attendee #                                                       4490 non-null   float64
 6   Group                                                            0 non-null      float64
 7   Order Type                                

In [5]:
train_missing_values = train_df.isnull().sum()
print("Missing values in training data:")
print(train_missing_values)

Missing values in training data:
First Name                                                            0
Email ID                                                              0
Quantity                                                              0
Price Tier                                                         4894
Ticket Type                                                           0
Attendee #                                                          404
Group                                                              4894
Order Type                                                            0
Currency                                                            404
Total Paid                                                            0
Fees Paid                                                           404
Eventbrite Fees                                                       0
Eventbrite Payment Processing                                         0
Attendee Status                

In [6]:
test_missing_values = test_df.isnull().sum()
print("Missing values in test data:")
print(test_missing_values)

Missing values in test data:
First Name                                                            0
Email ID                                                              0
Quantity                                                              0
Price Tier                                                         3796
Ticket Type                                                           0
Attendee #                                                            2
Group                                                              3796
Order Type                                                            0
Currency                                                              2
Total Paid                                                            0
Fees Paid                                                             2
Eventbrite Fees                                                       0
Eventbrite Payment Processing                                         0
Attendee Status                    

In [7]:
# We can fill missing values in "college_name" with "Unknown College"
train_df['College Name'].fillna('Unknown College', inplace=True)

In [8]:
# Since "placement_status" is the target variable, we cannot predict or fill missing values.
# Therefore, we will drop the rows with missing values in "placement_status."
train_df.dropna(subset=['Placement Status'], inplace=True)

In [9]:
# For "year of graduation," we can fill missing values with the median year
median_year_of_graduation = train_df['Year of Graduation'].median()
train_df['Year of Graduation'].fillna(median_year_of_graduation, inplace=True)

C:\Users\Naman\AppData\Roaming\Python\Python310\site-packages\numpy\lib\nanfunctions.py:1216: RuntimeWarning: Mean of empty slice
  return np.nanmean(a, axis, out=out, keepdims=keepdims)


In [10]:
print(train_df.isnull().sum())

First Name                                                            0
Email ID                                                              0
Quantity                                                              0
Price Tier                                                         1098
Ticket Type                                                           0
Attendee #                                                          402
Group                                                              1098
Order Type                                                            0
Currency                                                            402
Total Paid                                                            0
Fees Paid                                                           402
Eventbrite Fees                                                       0
Eventbrite Payment Processing                                         0
Attendee Status                                                 

In [11]:
test_df['College Name'].fillna('Unknown College', inplace=True)

In [12]:
test_df['Year of Graduation'].fillna(median_year_of_graduation, inplace=True)

In [13]:
print(test_df.isnull().sum())

First Name                                                            0
Email ID                                                              0
Quantity                                                              0
Price Tier                                                         3796
Ticket Type                                                           0
Attendee #                                                            2
Group                                                              3796
Order Type                                                            0
Currency                                                              2
Total Paid                                                            0
Fees Paid                                                             2
Eventbrite Fees                                                       0
Eventbrite Payment Processing                                         0
Attendee Status                                                 

In [14]:
target_col = "Placement Status"

In [70]:
train_df = pd.get_dummies(train_df, columns=['Ticket Type', 'College Name', 'Designation'])
test_df = pd.get_dummies(test_df, columns=['Ticket Type', 'College Name', 'Designation'])

In [71]:
print(train_df.isnull().sum())

First Name                           0
Email ID                             0
Quantity                             0
Price Tier                        4894
Attendee #                         404
                                  ... 
Designation_Software Engineer        0
Designation_Software developer       0
Designation_Sr. Manager              0
Designation_Students                 0
Designation_Web Development          0
Length: 140, dtype: int64


In [73]:
selected_features = ['Ticket Type', 'College Name', 'Designation', 'Year of Graduation', 'CGPA', 'Speaking Skills', 'ML Knowledge']
X_train = train_df[selected_features]
y_train = train_df[target_col]

In [20]:
# Perform one-hot encoding on categorical features
X_train_encoded = pd.get_dummies(X_train)

In [59]:
# Perform one-hot encoding on categorical features
X_test_encoded = pd.get_dummies(X_test)

In [82]:
# After encoding categorical features for both train and test data
X_train_encoded, X_test_encoded = X_train_encoded.align(X_test_encoded, join='left', axis=1)

In [83]:
from sklearn.impute import SimpleImputer

# Create an imputer
imputer = SimpleImputer(strategy='mean') 
X_train_imputed = imputer.fit_transform(X_train_encoded)
X_test_imputed = imputer.transform(X_test_encoded)

Model 1 : Logistic Regression

In [84]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression

In [18]:
model = LogisticRegression()

In [85]:
# Train the model on the imputed data
model.fit(X_train_imputed, y_train)

C:\Users\Naman\AppData\Roaming\Python\Python310\site-packages\sklearn\linear_model\_logistic.py:458: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


LogisticRegression()

In [86]:
from sklearn.model_selection import train_test_split

# Split the data into training and validation sets
X_train_split, X_val, y_train_split, y_val = train_test_split(X_train_imputed, y_train, test_size=0.2, random_state=42)

In [87]:
from sklearn.metrics import classification_report

# Predict on the validation set
y_val_pred = model.predict(X_val)

# Print classification report
print(classification_report(y_val, y_val_pred))


              precision    recall  f1-score   support

  Not placed       0.61      0.97      0.75       133
      Placed       0.50      0.05      0.08        87

    accuracy                           0.60       220
   macro avg       0.55      0.51      0.42       220
weighted avg       0.57      0.60      0.49       220



In [88]:
from sklearn.metrics import accuracy_score

# Calculate accuracy score
accuracy = accuracy_score(y_val, y_val_pred)

# Print the accuracy score
print("Accuracy:", accuracy)

Accuracy: 0.6045454545454545


Accuracy: 60.45%

Model 2: Support Vector Machine(SVM)

In [89]:
from sklearn.svm import SVC
from sklearn.metrics import classification_report, accuracy_score

In [90]:
# Create an SVM classifier
svm_model = SVC(kernel='linear', C=1)

In [91]:
# Train the model on the training data
svm_model.fit(X_train_imputed, y_train)

SVC(C=1, kernel='linear')

In [92]:
# Predict on the validation set
y_val_pred_svm = svm_model.predict(X_val)

In [93]:
# Calculate accuracy score
accuracy_svm = accuracy_score(y_val, y_val_pred_svm)

In [94]:
print(classification_report(y_val, y_val_pred_svm))

              precision    recall  f1-score   support

  Not placed       0.60      0.96      0.74       133
      Placed       0.29      0.02      0.04        87

    accuracy                           0.59       220
   macro avg       0.44      0.49      0.39       220
weighted avg       0.48      0.59      0.46       220



In [95]:
print("SVM Accuracy:", accuracy_svm)

SVM Accuracy: 0.5909090909090909


Accuracy:59.09%

Accuracy of logistic model is more than that of SVM model.
Therefore using logistic model for prediction of test data.

In [98]:
# Make predictions on test data
y_test_pred = model.predict(X_test_imputed)

In [99]:
# Save predictions to a DataFrame
predictions_df = pd.DataFrame({'Predicted Placement Status': y_test_pred})

In [101]:
predictions_df.value_counts()

Predicted Placement Status
Not placed                    3745
Placed                          51
dtype: int64

In [102]:
# Save predictions to an Excel file
predictions_df.to_excel('Predictions.xlsx', index=False)
print("Predictions saved to 'Predictions.xlsx'")

Predictions saved to 'Predictions.xlsx'
